File: frontend/services/ai.ts
import { apiRequest } from "./api";

export async function chatWithAI(message: string) {
  return await apiRequest(
    "/ai/chat",
    "POST",
    {
      message,
    }
  );
}

export async function summarizeText(text: string) {
  return await apiRequest(
    "/ai/summarize",
    "POST",
    {
      text,
    }
  );
}

export async function classifyIntent(text: string) {
  return await apiRequest(
    "/ai/classify",
    "POST",
    {
      text,
    }
  );
}

export async function extractEntities(text: string) {
  return await apiRequest(
    "/ai/entities",
    "POST",
    {
      text,
    }
  );
}

File: backend/app/routes/ai.py
"""
Bharat Sahayak AI
AI Router
"""

from pydantic import BaseModel
from fastapi import APIRouter

from app.services.ai_service import AIService

router = APIRouter(
    prefix="/ai",
    tags=["AI"],
)

ai_service = AIService()


class ChatRequest(BaseModel):
    message: str


@router.post("/chat")
def ai_chat(request: ChatRequest):
    return ai_service.generate_response(
        prompt=request.message
    )


@router.post("/summarize")
def summarize(request: ChatRequest):
    return {
        "summary": ai_service.summarize(
            request.message
        )
    }


@router.post("/classify")
def classify(request: ChatRequest):
    return ai_service.classify_intent(
        request.message
    )


@router.post("/entities")
def extract_entities(request: ChatRequest):
    return ai_service.extract_entities(
        request.message
    )